### DISCUSSION TITLE
Using `pathPrefix` in Node/SSR APIs returns both `pathPrefix` and `assetPrefix`

### DISCUSSION THREAD

**Discussion by mrseanbaines** [gatsbyjs/gatsby#34228] [Help]
### Preliminary Checks

- [X] This issue is not a duplicate. Before opening a new issue, please search existing issues: https://github.com/gatsbyjs/gatsby/issues
- [X] This issue is not a question, feature request, RFC, or anything other than a bug report directly related to Gatsby. Please post those things in GitHub Discussions: https://github.com/gatsbyjs/gatsby/discussions

### Description

When using the `pathPrefix` argument (e.g. `export const onRenderBody = ({ pathPrefix }) => {` which is provided in various Node/SSR APIs (e.g. `onPostBuild`, `onRenderBody` etc., the value contains both the `pathPrefix` _and_ `assetPrefix` defined in `gatsby-config`. Is this a bug? In contrast, when using the `withPrefix` helper imported from `gatsby`, the resulting value is only prefixed with the `pathPrefix` from `gatsby-config`.

Calling `withAssetPrefix` also prefixes with both `pathPrefix` _and_ `assetPrefix`. I'm not sure what the behaviour should be here.

```js
// gatsby-config.js
module.exports = {
  // ...
  assetPrefix: `/bananas`,
  pathPrefix: `/pyjamas`,
}
```

```js
// gatsby-ssr.js
import { withPrefix, withAssetPrefix } from "gatsby"

export const onRenderBody = ({ pathPrefix }) => {
  console.log("pathPrefix:", pathPrefix + "/")
  console.log("withPrefix:", withPrefix("/"))
  console.log("withAssetPrefix:", withAssetPrefix("/"))
}
```

```js
// results
pathPrefix: /bananas/pyjamas/
withPrefix: /pyjamas/
withAssetPrefix: /bananas/pyjamas/
```

### Reproduction Link

https://github.com/mrseanbaines/gatsby-prefix-bug

### Steps to Reproduce

1. Clone the project 
2. Install dependencies
3. Run `yarn build --prefix-paths`

### Expected Result

I would think that both console logs should print the same value, and both without the _asset_ prefix.

### Actual Result

The `pathPrefix` argument prints `/bananas/pyjamas/` and `withPrefix` prints `/pyjamas/`

### Environment

```shell
System:
  OS: macOS 12.0.1
  CPU: (8) x64 Intel(R) Core(TM) i5-8279U CPU @ 2.40GHz
  Shell: 5.8 - /bin/zsh
Binaries:
  Node: 14.17.1 - /var/folders/c6/zn3s5lkd09xf78n8tcylhyg00000gn/T/yarn--1637787313944-0.3128029059594468/node
  Yarn: 1.22.10 - /var/folders/c6/zn3s5lkd09xf78n8tcylhyg00000gn/T/yarn--1637787313944-0.3128029059594468/yarn
  npm: 7.19.0 - ~/.nvm/versions/node/v14.17.1/bin/npm
Languages:
  Python: 2.7.18 - /usr/bin/python
Browsers:
  Chrome: 96.0.4664.55
  Firefox: 94.0.2
  Safari: 15.1
npmPackages:
  gatsby: ^4.2.0 => 4.2.0
  gatsby-plugin-gatsby-cloud: ^4.2.0 => 4.2.0
  gatsby-plugin-image: ^2.2.0 => 2.2.0
  gatsby-plugin-manifest: ^4.2.0 => 4.2.0
  gatsby-plugin-offline: ^5.2.0 => 5.2.0
  gatsby-plugin-react-helmet: ^5.2.0 => 5.2.0
  gatsby-plugin-sharp: ^4.2.0 => 4.2.0
  gatsby-source-filesystem: ^4.2.0 => 4.2.0
  gatsby-transformer-sharp: ^4.2.0 => 4.2.0
```

**Comment by LekoArts** [comment]
Hi, thanks for the issue!

I understand that the current behavior can be confusing at times but it is the currently intended behavior. If we'd want to do changes to this it would need to happen in a new major version as it would be a breaking change (changes could be proposed here: https://github.com/gatsbyjs/gatsby/discussions/categories/rfc).

In https://github.com/gatsbyjs/gatsby/pull/34226 we clarified the language around this a little bit and also now mention the existence of `basePath`. It won't be accessible in `onRenderBody` at the moment as it's not passed.
